tidy3d/components/grid/grid_spec.py

```diff
@@ -263,49 +376,9 @@ def _make_coords_initial(
         else:
             bound_coords += self.custom_offset
 
-        # chop off any coords outside of simulation bounds, beyond some buffer region
-        # to take numerical effects into account
-        buffer = fp_eps * size
-        bound_min = center - size / 2 - buffer
-        bound_max = center + size / 2 + buffer
-
-        if bound_max < bound_coords[0] or bound_min > bound_coords[-1]:
-            axis_name = "xyz"[axis]
-            raise SetupError(
-                f"Simulation domain does not overlap with the provided custom grid in '{axis_name}' direction."
-            )
-
-        if size == 0:
-            # in case of zero-size dimension return the boundaries between which simulation falls
-            ind = np.searchsorted(bound_coords, center, side="right")
-
-            # in case when the center coincides with the right most boundary
-            if ind >= len(bound_coords):
-                ind = len(bound_coords) - 1
-
-            return bound_coords[ind - 1 : ind + 1]
-
-        else:
-            bound_coords = bound_coords[bound_coords <= bound_max]
-            bound_coords = bound_coords[bound_coords >= bound_min]
-
-            # if not extending to simulation bounds, repeat beginning and end
-            dl_min = dl[0]
-            dl_max = dl[-1]
-            while bound_coords[0] - dl_min >= bound_min:
-                bound_coords = np.insert(bound_coords, 0, bound_coords[0] - dl_min)
-            while bound_coords[-1] + dl_max <= bound_max:
-                bound_coords = np.append(bound_coords, bound_coords[-1] + dl_max)
-
-            # in case a `custom_offset` is provided, it's possible the bounds were numerically within
-            # the simulation bounds but were still chopped off, which is fixed here
-            if self.custom_offset is not None:
-                if np.isclose(bound_coords[0] - dl_min, bound_min):
-                    bound_coords = np.insert(bound_coords, 0, bound_coords[0] - dl_min)
-                if np.isclose(bound_coords[-1] + dl_max, bound_max):
-                    bound_coords = np.append(bound_coords, bound_coords[-1] + dl_max)
-
-            return bound_coords
+        return self._postprocess_unaligned_grid(
+            axis, structures, self.custom_offset is not None, bound_coords
```

Comment:
To increase readability `return self._postprocess_unaligned_grid(axis=axis, structures=structures, machine_error_relaxation=self.custom_offset is not None, bound_coords=bound_coords)`